src/main/java/io/cdap/plugin/kudu/KuduSinkConfig.java

```diff
@@ -260,4 +317,139 @@ public int getSeed() {
   public int getThreads() {
     return (optBossThreads != null) ? Integer.parseInt(optBossThreads) : 1;
   }
+
+  public void validate(FailureCollector failureCollector) {
+    if (Strings.isNullOrEmpty(optSchema)) {
+      failureCollector.addFailure("Write schema is not specified.", null)
+        .withConfigProperty(SCHEMA);
+    } else {
+      try {
+        Schema.parseJson(optSchema);
+      } catch (IOException e) {
+        failureCollector.addFailure( String.format("Unable to parse schema '%s'", optSchema),
+                                     "Provide correct schema.")
+          .withStacktrace(e.getStackTrace())
+          .withConfigProperty(SCHEMA);
+      }
+    }
+
+    // If there is macro specified for 'master' address or table name, then
+    // we defer the creation of table to initialize.
+    if (!containsMacro(MASTER)) {
+      if (Strings.isNullOrEmpty(optMasterAddresses)) {
+        failureCollector.addFailure("Kudu Master Server address list is empty.",
+                                    "Provide Kudu Master Server address list.")
+          .withConfigProperty(MASTER);
+      } else {
+        Arrays.stream(optMasterAddresses.split(",")).forEach(masterAddress -> {
+          if (masterAddress.split(":").length != 2){
+            failureCollector.addFailure(String.format("Master Address '%s' has incorrect format.", masterAddress),
+                                        "Provide Master Address in format - <hostname>:<port>")
```

Comment:
shouldn't it be `<hostname1>:<port1>,<hostname2>:<port2>,...`?

Also why some words are capitalized?

Reply:
<hostname1>:<port1>,<hostname2>:<port2>,... correct format
we check that each master address should be in format <hostname1>:<port1>